src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -134,20 +134,12 @@ export default function getAdaptationSwitchStrategy(
   }
 
   if (adaptation.type === "audio" &&
-      segmentBuffer.codec !== undefined &&
-      // We have been explicitly asked to reload
       options.audioTrackSwitchingMode === "direct" &&
-      // We're playing the current Period
-      isTimeInRange({ start, end }, currentTime) &&
-      // There is data for the current position or the codecs are differents
-      (playbackInfo.readyState > 1 || !hasCompatibleCodec(adaptation,
-                                                          segmentBuffer.codec)) &&
-      // We're not playing the current wanted audio Adaptation yet
-      !isTimeInRanges(adaptationInBuffer, currentTime))
-  {
-    return { type: "needs-reload", value: undefined };
+      isTimeInRange({ start, end }, currentTime)) {
+        return { type: "flush-buffer", value: [{ start, end }] }
```

Comment:
So here we're flushing only if we're playing this Period (start and end refer to the Period).
I don't remember what I told you last time (:P) but I though that for now we said that we flush regardless of whether this is the current Period

Also we're cleaning the whole period, not just the problematic segments (like in the clean-buffer case).
This could be OK but I wonder if there's no detail we miss (like removing too much data).

I'll be thinking a little about this one.